openlibrary/core/fulltext.py

```diff
@@ -76,3 +78,28 @@ def fulltext_search(q, page=1, limit=100, js=False, facets=False):
                 )
                 ia_results['hits']['hits'][idx]['availability'] = availability[ed.ocaid]
     return ia_results
+
+
+def ctx_func_factory():
+    """Factory function that returns setup_ctx_site for use with async_bridge.wrap
+    TODO: next time we use this in a different context, we should consider generalizing it.
+    """
+
+    forwarded_for = web.ctx.env.get('HTTP_X_FORWARDED_FOR', 'ol-internal')
+    user_agent = web.ctx.env.get('HTTP_USER_AGENT', '')
+
+    async def setup_env_vars():
+        from infogami.utils.delegate import create_site
+
+        web.ctx.site = create_site()
+        web.ctx.env = {
+            'HTTP_X_FORWARDED_FOR': forwarded_for,
+            'HTTP_USER_AGENT': user_agent,
```

Comment:
As noted in the other comment, let's try forwarding these explicitly into the appropriate fulltext method calls.